gvfs.h

```diff
@@ -14,7 +14,18 @@
 #define GVFS_SKIP_SHA_ON_INDEX                      (1 << 0)
 #define GVFS_BLOCK_COMMANDS                         (1 << 1)
 #define GVFS_MISSING_OK                             (1 << 2)
+
+/*
+ * This behavior of not deleting outside of the sparse-checkout
+ * is specific to the virtual filesystem support. It is only
+ * enabled by VFS for Git, and so can be used as an indicator
+ * that we are in a virtualized filesystem environment and not
+ * in a Scalar environment. This bit has two names to reflect
+ * that.
+ */
 #define GVFS_NO_DELETE_OUTSIDE_SPARSECHECKOUT       (1 << 3)
+#define GVFS_USE_VIRTUAL_FILESYSTEM                 (1 << 3)
```

Comment:
I could imagine that it might be better to rename `GVFS_NO_DELETE_OUTSIDE_SPARSECHECKOUT` to `GVFS_USE_VIRTUAL_FILESYSTEM` in a separate commit (fixing up, and amending the commit message of, 999461530634b67a79ce2027e407b6e6e2a9e5d1).

What do you think?

Speaking about commit messages... This part of the current commit's message only makes sense if the reader happens to have read the parent commit's message:

> This is also a highly requested option.

To age well, this sentence should be augmented to provide some context, e.g. "Just like the `--no-src` option that was added in the previous commit, support for `git worktree` in Scalar is also a highly requested feature."

Reply:
Commit message is updated to just say `This is a highly requested option.`.